src/briefcase/commands/build.py

```diff
@@ -123,19 +131,24 @@ def __call__(
         # and that the app configuration is finalized.
         self.finalize(app)
 
-        if app:
-            state = self._build_app(
-                app,
-                update=update,
-                update_requirements=update_requirements,
-                update_resources=update_resources,
-                update_support=update_support,
-                update_stub=update_stub,
-                no_update=no_update,
-                test_mode=test_mode,
-                **options,
-            )
+        if app_name:
+            # Build only the specified app
+            if app_name in self.apps:
+                state = self._build_app(
+                    self.apps[app_name],
```

Comment:
This logic doesn't seem right.

1. If you specify `app_name`, it will build *every* app, with the specified app's configuration - so the same app will be built N times.
2. If you *don't* specify `app_name` the fallback is to the default "build all apps", which ignores the case where `app` is passed in as an argument - for example, when you call `run -u`.